api/src/main/java/org/apache/iceberg/catalog/TableIdentifier.java

```diff
@@ -36,6 +36,7 @@
   private final String name;
 
   public static TableIdentifier of(String... names) {
+    Preconditions.checkArgument(names != null, "Table names must be non-null");
```

Comment:
I think this could be more clear because it sounds like a name may be null, when the problem is that the array is null. How about "Cannot create identifier from null array"? I'm actually not sure what a good error message would be here.

Reply:
good point. I think having `Cannot create table identifier from null array` is clear enough when this check fails.